src/arch/x86.rs

```diff
@@ -81,13 +84,20 @@ pub unsafe fn init(stack: &Stack, f: unsafe extern "C" fn(usize) -> !) -> StackP
         # When unwinding the frame corresponding to this function, a DWARF unwinder
         # will restore %ebx (and thus CFA of the first trampoline) from the stack slot.
         .cfi_offset %ebx, 4
+
+        # We start executing in the new stack by returning into this trampoline.
+        # The return address is set up to point after this nop so that the
+        # unwinder recognizes the caller as being this trampoline rather than
+        # whatever symbol is before this one.
+        nop
+
         # Push argument.
-        .cfi_def_cfa_offset 8
         pushl   %eax
+        .cfi_adjust_cfa_offset 4
         # Call the provided function.
-        call    *8(%esp)
+        call    *%ebp
```

Comment:
This makes the data flow even _more_ complicated with no benefit that I can see.